w_emitter.h

```diff
@@ -81,11 +81,26 @@ class WEmitter
         std::string call;
         get_functions(f, alloc_fcn, free_fcn, call);
         std::string other = ecall ? "enclave" : "host";
+        std::string fcn_id = edl_->name_ + "_fcn_id_" + f->name_;
 
         std::string args_t = f->name_ + "_args_t";
-        out() << prototype(f, ecall, gen_t(), prefix) << "{"
+        /*
+         * To avoid duplicated definition of the ecall wrapper on the host
+         * side, the ecall wrapper is defined as [edl_name]_[prefix]_[fun_name].
+         * Then we use weak_alias([edl_name]_[prefix]_[fun_name],
+         * [prefix]_[fun_name]) to make the exposed [prefix]_[fun_name] weak.
+         * Therefore, only one of the implementations will be picked by the
+         * linker.
+         */
+        std::string _prefix = ecall ? edl_->name_ + "_" + prefix : prefix;
```

Comment:
Seems like the prefix would also get applied to the ocall wrapper in _t.c. Is that the case?